### PR TITLE
{tools} [intel-2015a] NCO (REVIEW)

### PR DESCRIPTION
**Pull request by hajgato**
Needs PR ~~#1912~~ (netCDF 4.3.3.1, libxml2-2.9.2-Python-2.7.{9,10})

### CODE REVIEW

**Comment by hajgato** [comment]
Test report by @hajgato
**SUCCESS**
Build succeeded for 5 out of 5 (1 easyconfigs in this PR)
Linux RHEL 6.3, AMD Opteron(TM) Processor 6272, Python 2.6.6
See https://gist.github.com/a55a9d460d553ff96fdc for a full test report.

**Comment by hajgato** [comment]
Test report by @hajgato
**SUCCESS**
Build succeeded for 5 out of 5 (1 easyconfigs in this PR)
Linux SL 6.6, AMD Opteron(tm) Processor 6134, Python 2.6.6
See https://gist.github.com/e7842491d67c2888d361 for a full test report.

File: easybuild/easyconfigs/u/UDUNITS/UDUNITS-2.2.19-intel-2015a.eb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+dependencies = [('expat', '2.1.0')]
+
+sanity_check_paths = {
+    'files': [
+        'bin/udunits2',
+        'include/converter.h',
+        'include/udunits2.h',
+        'include/udunits.h',
+        'lib/libudunits2.a',
+        'lib/libudunits2.so',
```
Comment (boegel) [inline easybuild/easyconfigs/u/UDUNITS/UDUNITS-2.2.19-intel-2015a.eb]: hmm, is this too long when collapsed on one line? looks a bit stretched out like this...

```python
    'files': ['bin/udunits2', 'include/converter.h', 'include/udunits2.h', 'include/udunits.h',
              'lib/libudunits2.a', 'lib/libudunits2.so'],
...
```

File: easybuild/easyconfigs/a/ANTLR/ANTLR-2.7.7-intel-2015a.eb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+toolchain = {'name': 'intel', 'version': '2015a'}
+  
+source_urls = ['http://www.antlr2.org/download/']
+sources = [SOURCELOWER_TAR_GZ]
+
+patches = ['%(name)s-%(version)s_includes.patch']
+
+dependencies = [
+    ('Java', '1.8.0_31', '', True),
+    ('Python', '2.7.9'),
```
Comment (boegel) [inline easybuild/easyconfigs/a/ANTLR/ANTLR-2.7.7-intel-2015a.eb]: if Python is a dep, we usually reflect that via a versionsuffix...

also: 2.7.10?

File: easybuild/easyconfigs/l/libdap/libdap-3.14.0-intel-2015a.eb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+patches = ['%(name)s-%(version)s_flex.patch']
+
+builddependencies = [
+    ('Bison', '3.0.2'),
+    ('flex', '2.5.39'),
+]
+
+dependencies = [
+    ('cURL', '7.43.0'),
+    ('libxml2', '2.9.2', '-Python-2.7.9'),
```
Comment (boegel) [inline easybuild/easyconfigs/l/libdap/libdap-3.14.0-intel-2015a.eb]: 2.7.10?

**Comment by boegel** [comment]
@hajgato: did you see #1191? Just wondering if there's something in there that we missed here.
If this one goes in, I think we'll just close #1191 (or get it merged, if it's still relevant).

**Comment by hajgato** [comment]
It looks rather similar. I added `--disable-examples` to the ANTLR easyconfig (as it in PR #1191)
Only the toolchain and version differs (Plus PR #1191 does not have python dep for ANTLR)

**Comment by boegel** [comment]
Jenkins: test this please

File: easybuild/easyconfigs/a/ANTLR/ANTLR-2.7.7-intel-2015a-Python-2.7.10.eb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+source_urls = ['http://www.antlr2.org/download/']
+sources = [SOURCELOWER_TAR_GZ]
+
+patches = ['%(name)s-%(version)s_includes.patch']
+
+pythonversion = '2.7.10'
+versionsuffix = "-Python-%s" % pythonversion
+
+dependencies = [
+    ('Java', '1.8.0_31', '', True),
```
Comment (boegel) [inline easybuild/easyconfigs/a/ANTLR/ANTLR-2.7.7-intel-2015a-Python-2.7.10.eb]: bump to latest 1.8.0_60? (see http://www.oracle.com/technetwork/java/javase/downloads/index-jsp-138363.html#javasejdk)

File: easybuild/easyconfigs/a/ANTLR/ANTLR-2.7.7-intel-2015a-Python-2.7.9.eb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+source_urls = ['http://www.antlr2.org/download/']
+sources = [SOURCELOWER_TAR_GZ]
+
+patches = ['%(name)s-%(version)s_includes.patch']
+
+pythonversion = '2.7.9'
+versionsuffix = "-Python-%s" % pythonversion
+
+dependencies = [
+    ('Java', '1.8.0_31', '', True),
```
Comment (boegel) [inline easybuild/easyconfigs/a/ANTLR/ANTLR-2.7.7-intel-2015a-Python-2.7.9.eb]: 1.8.0_60?

File: easybuild/easyconfigs/l/libdap/libdap-3.14.0-intel-2015a-Python-2.7.10.eb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+patches = ['%(name)s-%(version)s_flex.patch']
+
+builddependencies = [
+    ('Bison', '3.0.2'),
+    ('flex', '2.5.39'),
+]
+
+pythonversion = '2.7.10'
+versionsuffix = "-Python-%s" % pythonversion
```
Comment (boegel) [inline easybuild/easyconfigs/l/libdap/libdap-3.14.0-intel-2015a-Python-2.7.10.eb]: `pythonversion` is used nowhere else, so just collapse it into `versionsuffix`, and move `versionsuffix` above right below `version`?

File: easybuild/easyconfigs/l/libdap/libdap-3.14.0-intel-2015a.eb (hunk 3 above)
Comment (boegel) [inline easybuild/easyconfigs/l/libdap/libdap-3.14.0-intel-2015a.eb]: use `versionsuffix` here too?

File: easybuild/easyconfigs/n/NCO/NCO-4.5.1-intel-2015a-Python-2.7.10.eb [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+source_urls = [SOURCEFORGE_SOURCE]
+sources = [SOURCELOWER_TAR_GZ]
+
+builddependencies = [
+    ('Bison', '3.0.2'),
+    ('flex', '2.5.39'),
+]
+
+pythonversion = '2.7.10'
+versionsuffix = "-Python-%s" % pythonversion
```
Comment (boegel) [inline easybuild/easyconfigs/n/NCO/NCO-4.5.1-intel-2015a-Python-2.7.10.eb]: collapse and move `versionsuffix` up?

File: easybuild/easyconfigs/n/NCO/NCO-4.5.1-intel-2015a.eb [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+source_urls = [SOURCEFORGE_SOURCE]
+sources = [SOURCELOWER_TAR_GZ]
+
+builddependencies = [
+    ('Bison', '3.0.2'),
+    ('flex', '2.5.39'),
+]
+
+dependencies = [
+    ('ANTLR', '2.7.7', '-Python-2.7.9'),
```
Comment (boegel) [inline easybuild/easyconfigs/n/NCO/NCO-4.5.1-intel-2015a.eb]: use `versionsuffix`?

**Comment by boegel** [comment]
@hajgato: some minor style remarks to tackle

is it worthwhile to have both the Python 2.7.9 and 2.7.10 easyconfigs?

if you don't need the 2.7.9 ones, just stick to 2.7.10?

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by hajgato** [comment]
@boegel done

File: easybuild/easyconfigs/l/LibUUID/LibUUID-1.0.3-intel-2015a.eb [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+name = 'LibUUID'
+version = '1.0.3'
+
+homepage = 'http://sourceforge.net/projects/libuuid/'
+description = """Portable uuid C library"""
+
+source_urls  = [SOURCEFORGE_SOURCE]
+sources = [SOURCELOWER_TAR_GZ]
+
+toolchain = {'name': 'intel', 'version': '2015a'}
```
Comment (boegel) [inline easybuild/easyconfigs/l/LibUUID/LibUUID-1.0.3-intel-2015a.eb]: @hajgato: please move up above `source_urls`?

File: easybuild/easyconfigs/l/libdap/libdap-3.14.0-intel-2015a-Python-2.7.10.eb [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+homepage = 'http://opendap.org/download/libdap'
+description = """A C++ SDK which contains an implementation of DAP 2.0
+ and the development versions of DAP3, up to 3.4.
+ This includes both Client- and Server-side support classes."""
+
+source_urls  = ['http://www.opendap.org/pub/source/']
+sources = [SOURCE_TAR_GZ]
+
+toolchain = {'name': 'intel', 'version': '2015a'}
```
Comment (boegel) [inline easybuild/easyconfigs/l/libdap/libdap-3.14.0-intel-2015a-Python-2.7.10.eb]: above `source_urls` pls

File: easybuild/easyconfigs/a/ANTLR/ANTLR-2.7.7-intel-2015a-Python-2.7.10.eb [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+source_urls = ['http://www.antlr2.org/download/']
+sources = [SOURCELOWER_TAR_GZ]
+
+patches = ['%(name)s-%(version)s_includes.patch']
+
+pythonversion = '2.7.10'
+versionsuffix = "-Python-%s" % pythonversion
+
+dependencies = [
+    ('Java', '1.8.0_60', '', True),
```
Comment (boegel) [inline easybuild/easyconfigs/a/ANTLR/ANTLR-2.7.7-intel-2015a-Python-2.7.10.eb]: @hajgato: missing easyconfig file for this one?

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 4 out of 6 (6 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/34a24d918909bce2e1d0 for a full test report.

**Comment by boegel** [comment]
@hajgato: ignore the failed test report, download fail because of `ftp://`

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/cb545f4f25c1feec7718 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (6 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/acaaacca6564799d01ea for a full test report.

**Comment by boegel** [comment]
looks good, will wait until Jenkins can give us a green light

**Comment by hajgato** [comment]
@boegel can you trigger Jenkins?

**Comment by hajgato** [comment]
@boegel: this was only waiting for Jenkins' green light

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/f494f70fc2215d34ff27 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 9 out of 9 (6 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/b4cc906524ddb1860d8c for a full test report.

**Comment by boegel** [comment]
Going in, thanks @hajgato!

**Pull request merged**